packages/nimble-components/src/table-column/text/tests/table-column-text.spec.ts

```diff
@@ -0,0 +1,197 @@
+        element.data = [...simpleTableData];
+        await connect();
+        await waitForUpdatesAsync();
+
+        const firstColumn = element.columns[0] as TableColumnText;
+        firstColumn.placeholder = 'different value';
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(3, 0)).toBe('different value');
+    });
+
+    it('changing data from value to null displays placeholder', async () => {
+        element.data = [...simpleTableData];
+        await connect();
+        await waitForUpdatesAsync();
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('string 1');
+
+        const updatedValue = { field: null };
+        const updatedData = [updatedValue];
+        element.data = updatedData;
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('no value');
+    });
+
+    it('changing data from null to value displays value', async () => {
+        element.data = [...simpleTableData];
+        await connect();
+        await waitForUpdatesAsync();
+        expect(pageObject.getRenderedCellContent(3, 0)).toBe('no value');
+
+        const updatedData: SimpleTableRecord[] = [
+            simpleTableData[0],
+            simpleTableData[1],
+            simpleTableData[2],
+            { field: 'new value' }
+        ];
+        element.data = updatedData;
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(3, 0)).toBe('new value');
+    });
+
+    it('when no fieldName provided, nothing is displayed', async () => {
+        await connect();
+        await waitForUpdatesAsync();
+
+        const firstColumn = element.columns[0] as TableColumnText;
+        firstColumn.fieldName = undefined;
+        element.data = [...simpleTableData];
+        await waitForUpdatesAsync();
+
+        expect(pageObject.getRenderedCellContent(0, 0)).toBe('');
+    });
+
+    const fieldValues = [
+        { dataValue: 'foo', renderedValue: 'foo' },
+        { dataValue: '<button></button>', renderedValue: '<button></button>' },
+        { dataValue: null, renderedValue: 'no value' },
+        { dataValue: 'null', renderedValue: 'null' },
+        { dataValue: undefined, renderedValue: 'no value' },
+        { dataValue: 'undefined', renderedValue: 'undefined' }
+    ];
+    for (const fieldValue of fieldValues) {
+        let testDataTitleValue: string;
+        if (fieldValue.dataValue) {
+            testDataTitleValue = `"${fieldValue.dataValue}"`;
+        } else if (fieldValue.dataValue === null) {
+            testDataTitleValue = 'null';
+        } else {
+            testDataTitleValue = 'undefined';
+        }
+        // eslint-disable-next-line @typescript-eslint/no-loop-func
+        it(`data ${testDataTitleValue} renders as ${fieldValue.renderedValue}`, async () => {
+            await connect();
+
+            const updatedValue = { field: fieldValue.dataValue };
```

Comment:
I got a bit thrown off by the term `updated` here -- both in `updatedValue` and `updatedData` since it is the initial data set on the table. What if you consolidate these three lines into:
```ts
element.data = [{ field: fieldValue.data }];
```

Reply:
Done (duh).